vasyl_varyshchuk/Week_1/task_3/generate_file_size.py

```diff
@@ -2,46 +2,49 @@
 This module returns random size of file
 """
 
+import os
 import random
 
 list_of_files = ['reports.txt', 'monthly_reports.zip', 'README.txt',
                  'picture.jpg', 'logo.ttf', 'config.ini']
 
 
-def recognize_file_type(a_file):
-    return a_file.split('.')[1]
+def recognize_file_type(file_name):
+    _, extension = os.path.splitext(file_name)
+    return extension
 
 
-def run_executors(file_name):
-    def txt_handler(file_ext):
-        return random.randint(100, 450) if file_ext == 'txt' else None
+def txt_handler():
+    return random.randint(100, 450)
 
-    def zip_handler(file_extension):
-        return random.randint(500, 700) if file_extension == 'zip' else None
 
-    def jpg_handler(file_extension):
-        return random.randint(800, 1000) if file_extension == 'jpg' else None
+def zip_handler():
+    return random.randint(500, 700)
 
-    def ttf_handler(file_extension):
-        return random.randint(1100, 1300) if file_extension == 'ttf' else None
 
-    def ini_handler(file_extension):
-        return random.randint(1300, 1500) if file_extension == 'ini' else None
+def jpg_handler():
+    return random.randint(800, 1000)
 
-    ext = recognize_file_type(file_name)
 
-    functions = {
-        'txt': txt_handler,
-        'zip': zip_handler,
-        'jpg': jpg_handler,
-        'ttf': ttf_handler,
-        'ini': ini_handler
-    }
+def ttf_handler():
+    return random.randint(1100, 1300)
+
 
-    choosed_func = functions[ext]
-    size = choosed_func(ext)
+def ini_handler():
+        return random.randint(1300, 1500)
 
-    print('Generated size of .{} file => {}'.format(ext, size))
+
+def run_executors(file_name):
+    extension = recognize_file_type(file_name)
+    functions = {
```

Comment:
You should move this mapping from the function.

Reply:
This map has already moved out from function and the changes were committed.